File: Data Structures & Algorithms/string-encode-and-decode/submission-1.java
class Solution {

    public String encode(List<String> strs) {
        String solution = "";
        
        for (String s : strs) {
            solution += "'" + Integer.toString(s.length()) + "\"" + s;
        }
        return solution;
    }

    public List<String> decode(String str) {
        List<String> arr = new ArrayList<>();

        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) == '\'') {
                int count = i + 1;
                while (str.charAt(count) != '\"') {
                    count++;
                }
                int length = Integer.parseInt(str.substring(i + 1, count));

                String toAdd = str.substring(count + 1, count + length + 1);
                arr.add(toAdd);
            }
        }
        return arr;
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        arr = [0] * (n + 1)
        for i in range(n, -1, -1):
            if i == n or i == n - 1:
                arr[i] = 1
                continue
            arr[i] = arr[i + 1] + arr[i + 2]
        return arr[0]